pkg/mdformatter/mdformatter.go

```diff
@@ -109,11 +110,29 @@ func (FormatFrontMatter) TransformFrontMatter(_ SourceContext, frontMatter map[s
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(keys)))
 
-	b := bytes.NewBuffer([]byte("---\n"))
+	b := bytes.NewBuffer([]byte("---"))
 	for _, k := range keys {
-		_, _ = fmt.Fprintf(b, "%v: %v\n", k, frontMatter[k])
+		// Check if frontMatter[k] is a map.
+		frontMatterStr := fmt.Sprintf("%s", frontMatter[k])
+		_, isMap := frontMatter[k].(map[string]interface{})
```

Comment:
Actually we can do better. Since you are casting to map we could use this map directly, no need to use `frontMatterStr`, no? (:

Reply:
Oh yes. Will make the changes!

Reply:
Used the casted map. Code looks better now!